_posts/2019-12-20-differential-privacy/differential_privacy.Rmd

```diff
@@ -0,0 +1,470 @@
+
+Intuitively paraphrasing results from a domain where results are communicated as theorems and proofs [@dwork2006differential][@Dwork], the only achievable (in a lossy but quantifiable way) objective is that from queries to a database, nothing more should be learned about an individual in that database than if they hadn't been in there at all.[@primer] 
+
+What this statement does is caution against overly high expectations: Even if query results are reported in a DP way (we'll see how that goes in a second), they enable some probabilistic inferences about individuals in the respective population. (Otherwise, why conduct studies at all.)
+
+So how is DP being achieved? The main ingredient is _noise_ added to the results of a query. In the above cat example, instead of exact numbers we'd report approximate ones: "Of ~ 100 cats living in Y, about 30 are overweight...". If this is done for both of the above studies, no inference will be possible about aunt G.'s cat.
+
+Even with random noise added to query results though, answers to _repeated_ queries will leak information. So in reality, there is a _privacy budget_ that can be tracked, and may be used up in the course of consecutive queries.
+
+This is reflected in the formal definition of DP. The idea is that queries to two databases differing in at most one element should give basically the same result. Put formally [@dwork2006differential]:
+
+> A randomized function $\mathcal{K}$ gives $\epsilon$ -differential privacy if for all data sets D1 and D2 differing on at most one element, and all $S \subseteq Range(K)$, 
+
+> $Pr[\mathcal{K}(D1)\in S] \leq exp(\epsilon) × Pr[K(D2) \in S]$
+
+This $\epsilon$ -differential privacy is additive: If one query is $\epsilon$-DP at a value of 0.01, and another one at 0.03, together they will be 0.04 $\epsilon$-differentially private.
+
+If $\epsilon$-DP is to be achieved via adding noise, how exactly should this be done? Here, several mechanisms exist; the basic, intuitively plausible principle though is that the amount of noise should be calibrated to the target function's sensitivity, defined as the maximum $l1$ norm of the difference of function values computed on all pairs of datasets differing in a single example.
+
+So far, we've been talking about databases and datasets. How does this apply to machine and/or deep learning? 
+
+
+
+## TensorFlow Privacy
+
+Applying DP to deep learning, we want a model's parameters to wind up "essentially the same" whether trained on a dataset including that cute little kitty or not. TensorFlow (TF) Privacy [@abaditfp], a library built on top of TF, makes it easy on users to add privacy guarantees to their models -- easy, that is, from a technical point of view. (As with life overall, the hard decisions on _how much_ of an asset we should be reaching for, and how to trade off one asset (here: privacy) with another (here: model performance), remain to be taken by each of us ourselves.)
```

Comment:
```suggestion
Applying DP to deep learning, we want a model's parameters to wind up "essentially the same" whether trained on a dataset including that cute little kitty or not. TensorFlow (TF) Privacy [@abaditfp], a library built on top of TF, makes it easy for users to add privacy guarantees to their models -- easy, that is, from a technical point of view. (As with life overall, the hard decisions on _how much_ of an asset we should be reaching for, and how to trade off one asset (here: privacy) with another (here: model performance), remain to be taken by each of us ourselves.)
```

Reply:
Changed "easy on users" to "easy for users" (IDK why it doesn't show up in the diff).